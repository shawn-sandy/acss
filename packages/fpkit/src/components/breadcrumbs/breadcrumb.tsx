// Code: Breadcrumb component
import React from "react";
import UI from "#components/ui";
import { Truncate } from "#libs/content";
import Link from "#components/link/link";

// ============================================================================
// TYPES
// ============================================================================

/**
 * Represents a route segment in the breadcrumb navigation.
 *
 * @remarks
 * Each route can customize its display name and URL independently from its path.
 * This allows for URL aliasing and custom route naming.
 *
 * @example
 * ```tsx
 * const route: CustomRoute = {
 *   path: "prod",
 *   name: "Products",
 *   url: "/products"
 * };
 * ```
 */
export type CustomRoute = {
  /** The path segment as it appears in the URL */
  path?: string;
  /** The display name shown to users */
  name: string;
  /** The URL for navigation (defaults to path if not provided) */
  url?: string;
};

/**
 * Props for the Breadcrumb component.
 *
 * @remarks
 * The component can operate in two modes:
 * 1. Automatic mode: Derives path from `currentRoute` prop
 * 2. Controlled mode: Uses provided `routes` array for complete control over route naming
 *
 * @example
 * ```tsx
 * // Simple automatic mode
 * <Breadcrumb currentRoute="/products/shirts" />
 *
 * // Controlled mode with custom route names
 * <Breadcrumb
 *   currentRoute="/prod/shirts"
 *   routes={[
 *     { path: "prod", name: "Products", url: "/products" },
 *     { path: "shirts", name: "All Shirts", url: "/products/shirts" }
 *   ]}
 * />
 * ```
 */
export type BreadcrumbProps = {
  /** Array of custom route objects for controlled breadcrumb generation */
  routes?: CustomRoute[];
  /** Starting route node (typically "Home") */
  startRoute?: React.ReactNode;
  /** Starting route URL (typically "/") */
  startRouteUrl?: string;
  /** Separator element between breadcrumb items */
  spacer?: React.ReactNode;
  /** Current route path (required for breadcrumb generation) */
  currentRoute?: string;
  /** ARIA label for the breadcrumb navigation */
  ariaLabel?: string;
  /** Maximum character length before truncating breadcrumb text */
  truncateLength?: number;
  /** Props to spread onto breadcrumb Link components */
  linkProps?: Omit<React.ComponentProps<typeof Link>, "href" | "children">;
} & Omit<React.ComponentProps<typeof UI>, "as" | "aria-label">;

// ============================================================================
// SUB-COMPONENTS
// ============================================================================

/**
 * BreadcrumbItem - Individual list item wrapper for breadcrumb segments.
 *
 * @remarks
 * This is a presentational component that wraps each breadcrumb segment.
 * Memoized to prevent unnecessary re-renders when parent updates.
 */
const BreadcrumbItem = React.memo(
  ({
    children,
    id,
    styles,
    classes,
    ...props
  }: React.ComponentProps<typeof UI>) => {
    return (
      <li
        id={id}
        style={styles}
        className={classes}
        data-list="unstyled inline"
        {...props}
      >
        {children}
      </li>
    );
  }
);
BreadcrumbItem.displayName = "BreadcrumbItem";

/**
 * BreadcrumbList - Ordered list container for breadcrumb items.
 *
 * @remarks
 * Uses semantic `<ol>` element as recommended by WCAG for breadcrumb navigation.
 * Memoized to prevent unnecessary re-renders.
 */
const BreadcrumbList = React.memo(
  ({ children, ...props }: React.ComponentProps<typeof UI>) => {
    return (
      <UI as="ol" data-list="unstyled inline" {...props}>
        {children}
      </UI>
    );
  }
);
BreadcrumbList.displayName = "BreadcrumbList";

/**
 * BreadcrumbNav - Navigation wrapper for breadcrumb structure.
 *
 * @remarks
 * Provides semantic `<nav>` element with proper ARIA labeling for screen readers.
 * Automatically wraps children in BreadcrumbList.
 */
const BreadcrumbNav = React.memo(
  ({
    styles,
    id,
    classes,
    children,
    ...props
  }: React.ComponentProps<typeof UI>) => {
    return (
      <UI as="nav" id={id} styles={styles} className={classes} {...props}>
        <BreadcrumbList>{children}</BreadcrumbList>
      </UI>
    );
  }
);
BreadcrumbNav.displayName = "BreadcrumbNav";

// ============================================================================
// HOOKS
// ============================================================================

/**
 * Custom hook to process breadcrumb segments from a path string.
 *
 * @param currentRoute - The current route path to process
 * @param routes - Optional custom route mappings
 * @returns Processed breadcrumb segments with metadata
 *
 * @remarks
 * This hook encapsulates the business logic for:
 * - Path parsing and segmentation
 * - Route name resolution from custom routes
 * - URL construction for each segment
 *
 * Memoized to prevent unnecessary recalculations on each render.
 */
function useBreadcrumbSegments(
  currentRoute: string | undefined,
  routes?: CustomRoute[]
) {
  const segments = React.useMemo(() => {
    if (!currentRoute) return [];
    return currentRoute.split("/").filter((segment) => segment);
  }, [currentRoute]);

  const getRouteMetadata = React.useCallback(
    (pathSegment: string): CustomRoute => {
      const route = routes?.find((r) => r.path === pathSegment);

      return {
        path: route?.path || pathSegment,
        name: route?.name || pathSegment,
        url: route?.url || pathSegment,
      };
    },
    [routes]
  );

  const processedSegments = React.useMemo(() => {
    return segments.map((segment, index) => ({
      ...getRouteMetadata(segment),
      isLast: index === segments.length - 1,
      index,
    }));
  }, [segments, getRouteMetadata]);

  return {
    segments: processedSegments,
    hasSegments: processedSegments.length > 0,
  };
}

// ============================================================================
// MAIN COMPONENT
// ============================================================================

/**
 * Breadcrumb - Navigation component displaying hierarchical page location.
 *
 * @remarks
 * A WCAG 2.1 AA compliant breadcrumb navigation component that helps users
 * understand their current location within a site hierarchy and navigate back
 * to parent pages.
 *
 * ## Features
 * - Automatic path parsing from `currentRoute` prop
 * - Custom route naming via `routes` array
 * - Text truncation for long route names
 * - Full accessibility support with ARIA attributes
 * - Performance optimized with memoization
 *
 * ## Accessibility
 * - Uses semantic `<nav>` and `<ol>` elements
 * - Proper `aria-label` for screen reader context
 * - Current page marked with `aria-current="page"`
 * - Decorative separators hidden from screen readers with `aria-hidden="true"`
 * - Truncated text includes full text in `aria-label`
 *
 * @example
 * ```tsx
 * // Basic usage
 * <Breadcrumb currentRoute="/products/shirts/blue-shirt" />
 * // Renders: Home / products / shirts / blue-shirt
 *
 * // With custom route names
 * <Breadcrumb
 *   currentRoute="/products/shirts/item-123"
 *   routes={[
 *     { path: "products", name: "All Products", url: "/products" },
 *     { path: "shirts", name: "Shirts & Tops", url: "/products/shirts" },
 *     { path: "item-123", name: "Blue Cotton Shirt", url: "/products/shirts/item-123" }
 *   ]}
 * />
 * // Renders: Home / All Products / Shirts & Tops / Blue Cotton Shirt
 *
 * // With custom starting point and styling
 * <Breadcrumb
 *   currentRoute="/about/team"
 *   startRoute="Dashboard"
 *   startRouteUrl="/dashboard"
 *   spacer={<span> â†’ </span>}
 *   ariaLabel="Page navigation"
 *   truncateLength={20}
 * />
 * ```
 *
 * @param props - Component props
 * @returns Breadcrumb navigation element or null if no valid route
 */
export const Breadcrumb = ({
  startRoute = "Home",
  startRouteUrl = "/",
  currentRoute,
  spacer = <>&#47;</>,
  routes,
  styles,
  id,
  classes,
  ariaLabel = "Breadcrumb",
  truncateLength = 15,
  linkProps,
  ...props
}: BreadcrumbProps): React.JSX.Element | null => {
  const { segments, hasSegments } = useBreadcrumbSegments(currentRoute, routes);
  const uuid = React.useId();

  // Early return if no valid path
  if (!currentRoute?.length || !hasSegments) {
    return null;
  }

  return (
    <BreadcrumbNav
      id={id}
      styles={styles}
      className={classes}
      aria-label={ariaLabel}
      {...props}
    >
      {/* Home/Start Route */}
      <BreadcrumbItem key={`start-${uuid}`}>
        <Link href={startRouteUrl} {...linkProps}>
          {startRoute}
        </Link>
      </BreadcrumbItem>

      {/* Path Segments */}
      {segments.map(({ name, url, path, isLast, index }) => {
        const decodedName = decodeURIComponent(name);
        const truncatedName = Truncate(decodedName, truncateLength);
        const needsAriaLabel = decodedName.length > truncateLength;

        // Current page (last segment)
        if (isLast) {
          // Skip if segment is too short or duplicate of previous
          const previousPath = index > 0 ? segments[index - 1].path : null;
          if (!path || path.length <= 3 || path === previousPath) {
            return null;
          }

          return (
            <BreadcrumbItem key={`${path}-${uuid}`}>
              <span aria-hidden="true">{spacer}</span>
              <span
                aria-current="page"
                aria-label={needsAriaLabel ? decodedName : undefined}
              >
                {truncatedName}
              </span>
            </BreadcrumbItem>
          );
        }

        // Intermediate segments (links)
        return (
          <BreadcrumbItem key={`${path}-${uuid}`}>
            <span aria-hidden="true">{spacer}</span>
            <Link
              href={url}
              aria-label={needsAriaLabel ? decodedName : undefined}
              {...linkProps}
            >
              {truncatedName}
            </Link>
          </BreadcrumbItem>
        );
      })}
    </BreadcrumbNav>
  );
};

// ============================================================================
// EXPORTS
// ============================================================================

export default Breadcrumb;

Breadcrumb.displayName = "Breadcrumb";
Breadcrumb.Nav = BreadcrumbNav;
Breadcrumb.List = BreadcrumbList;
Breadcrumb.Item = BreadcrumbItem;
