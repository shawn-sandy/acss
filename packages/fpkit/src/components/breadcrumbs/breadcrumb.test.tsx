import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Breadcrumb, CustomRoute } from "./breadcrumb";

describe("Breadcrumb", () => {
  // ============================================================================
  // CORE FUNCTIONALITY TESTS
  // ============================================================================

  describe("Core Functionality", () => {
    it("renders a navigation element with breadcrumb list", () => {
      render(<Breadcrumb currentRoute="/products/shirts" />);
      const nav = screen.getByRole("navigation");
      expect(nav).toBeInTheDocument();

      const list = nav.querySelector("ol");
      expect(list).toBeInTheDocument();
    });

    it("returns null when currentRoute is undefined", () => {
      const { container } = render(<Breadcrumb currentRoute={undefined} />);
      expect(container.firstChild).toBeNull();
    });

    it("returns null when currentRoute is empty string", () => {
      const { container } = render(<Breadcrumb currentRoute="" />);
      expect(container.firstChild).toBeNull();
    });

    it("renders home/start route link correctly", () => {
      render(<Breadcrumb currentRoute="/products" />);
      const homeLink = screen.getByRole("link", { name: "Home" });
      expect(homeLink).toBeInTheDocument();
      expect(homeLink).toHaveAttribute("href", "/");
    });

    it("parses path segments correctly from currentRoute", () => {
      render(<Breadcrumb currentRoute="/products/shirts/blue-shirt" />);

      expect(screen.getByText("products")).toBeInTheDocument();
      expect(screen.getByText("shirts")).toBeInTheDocument();
      expect(screen.getByText("blue-shirt")).toBeInTheDocument();
    });

    it("handles paths with leading slashes", () => {
      render(<Breadcrumb currentRoute="/products" />);
      expect(screen.getByText("products")).toBeInTheDocument();
    });

    it("handles paths with trailing slashes", () => {
      render(<Breadcrumb currentRoute="/products/shirts/" />);
      expect(screen.getByText("products")).toBeInTheDocument();
      expect(screen.getByText("shirts")).toBeInTheDocument();
    });

    it("handles encoded URI components correctly", () => {
      render(<Breadcrumb currentRoute="/products/learning%20in%20public" />);

      expect(screen.getByText("products")).toBeInTheDocument();

      // Text is truncated, so use aria-label to verify full decoded text
      const element = screen.getByText(/learning in pub/);
      expect(element).toHaveAttribute("aria-label", "learning in public");
    });
  });

  // ============================================================================
  // CUSTOM ROUTES TESTS
  // ============================================================================

  describe("Custom Routes", () => {
    const customRoutes: CustomRoute[] = [
      {
        path: "products",
        name: "All Products",
        url: "/products",
      },
      {
        path: "shirts",
        name: "Shirts & Tops",
        url: "/products/shirts",
      },
    ];

    it("maps path segments to custom route names", () => {
      render(
        <Breadcrumb
          currentRoute="/products/shirts"
          routes={customRoutes}
        />
      );

      expect(screen.getByText("All Products")).toBeInTheDocument();
      expect(screen.getByText("Shirts & Tops")).toBeInTheDocument();
    });

    it("uses custom URLs when provided", () => {
      render(
        <Breadcrumb
          currentRoute="/products/shirts"
          routes={customRoutes}
        />
      );

      const productsLink = screen.getByRole("link", { name: "All Products" });
      expect(productsLink).toHaveAttribute("href", "/products");

      const shirtsSpan = screen.getByText("Shirts & Tops");
      expect(shirtsSpan).toBeInTheDocument();
    });

    it("falls back to path segment when no custom route found", () => {
      render(
        <Breadcrumb
          currentRoute="/products/pants"
          routes={customRoutes}
        />
      );

      expect(screen.getByText("All Products")).toBeInTheDocument();
      expect(screen.getByText("pants")).toBeInTheDocument(); // No custom route
    });

    it("handles partial custom route mappings", () => {
      const partialRoutes: CustomRoute[] = [
        {
          path: "products",
          name: "Products",
          url: "/products",
        },
      ];

      render(
        <Breadcrumb
          currentRoute="/products/shirts/item-123"
          routes={partialRoutes}
        />
      );

      expect(screen.getByText("Products")).toBeInTheDocument();
      expect(screen.getByText("shirts")).toBeInTheDocument();
      expect(screen.getByText("item-123")).toBeInTheDocument();
    });
  });

  // ============================================================================
  // ACCESSIBILITY TESTS
  // ============================================================================

  describe("Accessibility", () => {
    it("renders semantic nav element with proper aria-label", () => {
      render(
        <Breadcrumb
          currentRoute="/products"
          ariaLabel="Page navigation"
        />
      );

      const nav = screen.getByRole("navigation", { name: "Page navigation" });
      expect(nav).toBeInTheDocument();
    });

    it("uses default aria-label of 'Breadcrumb' when not provided", () => {
      render(<Breadcrumb currentRoute="/products" />);

      const nav = screen.getByRole("navigation", { name: "Breadcrumb" });
      expect(nav).toBeInTheDocument();
    });

    it("uses ordered list (ol) for breadcrumb list", () => {
      render(<Breadcrumb currentRoute="/products/shirts" />);

      const nav = screen.getByRole("navigation");
      const list = nav.querySelector("ol");
      expect(list).toBeInTheDocument();
    });

    it("marks last segment with aria-current='page'", () => {
      render(<Breadcrumb currentRoute="/products/shirts" />);

      const currentPage = screen.getByText("shirts");
      expect(currentPage).toHaveAttribute("aria-current", "page");
    });

    it("hides spacers from screen readers with aria-hidden", () => {
      render(<Breadcrumb currentRoute="/products/shirts" />);

      const nav = screen.getByRole("navigation");
      const spacers = nav.querySelectorAll('[aria-hidden="true"]');

      // Should have at least 2 spacers (one for each segment)
      expect(spacers.length).toBeGreaterThan(0);
    });

    it("does not render anchor tags with href='#'", () => {
      render(<Breadcrumb currentRoute="/products/shirts" />);

      const nav = screen.getByRole("navigation");
      const invalidLinks = nav.querySelectorAll('a[href="#"]');

      expect(invalidLinks.length).toBe(0);
    });

    it("provides full text in aria-label when truncated", () => {
      const longName = "this-is-a-very-long-product-name";
      render(
        <Breadcrumb
          currentRoute={`/products/${longName}`}
          truncateLength={10}
        />
      );

      const truncatedElement = screen.getByText(/this-is-a-/);
      expect(truncatedElement).toHaveAttribute("aria-label", longName);
    });

    it("does not add aria-label when text is not truncated", () => {
      render(
        <Breadcrumb
          currentRoute="/products/shirt"
          truncateLength={20}
        />
      );

      const element = screen.getByText("shirt");
      expect(element).not.toHaveAttribute("aria-label");
    });
  });

  // ============================================================================
  // TRUNCATION TESTS
  // ============================================================================

  describe("Text Truncation", () => {
    it("truncates text beyond default truncateLength (15)", () => {
      const longName = "verylongproductname";
      render(<Breadcrumb currentRoute={`/products/${longName}`} />);

      // Should be truncated to 15 chars + "..."
      expect(screen.getByText("verylongproduct...")).toBeInTheDocument();
    });

    it("respects custom truncateLength prop", () => {
      const longName = "verylongname";
      render(
        <Breadcrumb
          currentRoute={`/products/${longName}`}
          truncateLength={5}
        />
      );

      expect(screen.getByText("veryl...")).toBeInTheDocument();
    });

    it("does not truncate text shorter than truncateLength", () => {
      render(
        <Breadcrumb
          currentRoute="/products/shirt"
          truncateLength={15}
        />
      );

      expect(screen.getByText("shirt")).toBeInTheDocument();
    });

    it("truncates both intermediate and current page segments", () => {
      render(
        <Breadcrumb
          currentRoute="/verylongfirstsegment/verylongsecondsegment"
          truncateLength={10}
        />
      );

      expect(screen.getByText("verylongfi...")).toBeInTheDocument();
      expect(screen.getByText("verylongse...")).toBeInTheDocument();
    });
  });

  // ============================================================================
  // EDGE CASES TESTS
  // ============================================================================

  describe("Edge Cases", () => {
    it("skips last segment if length <= 3 characters", () => {
      render(<Breadcrumb currentRoute="/products/shirts/abc" />);

      // "abc" should be skipped (last segment with length 3)
      expect(screen.queryByText("abc")).not.toBeInTheDocument();
      expect(screen.getByText("products")).toBeInTheDocument();
      expect(screen.getByText("shirts")).toBeInTheDocument();
    });

    it("skips duplicate consecutive segments", () => {
      const { container } = render(
        <Breadcrumb currentRoute="/products/products" />
      );

      const nav = container.querySelector("nav");
      const productLinks = nav?.querySelectorAll('a[href="products"]');

      // Should only have one "products" link (first occurrence)
      expect(productLinks?.length).toBeLessThanOrEqual(1);
    });

    it("handles single segment path", () => {
      render(<Breadcrumb currentRoute="/about" />);

      expect(screen.getByRole("link", { name: "Home" })).toBeInTheDocument();
      expect(screen.getByText("about")).toBeInTheDocument();
    });

    it("handles deep nesting (many segments)", () => {
      render(
        <Breadcrumb currentRoute="/level1/level2/level3/level4/level5" />
      );

      expect(screen.getByText("level1")).toBeInTheDocument();
      expect(screen.getByText("level2")).toBeInTheDocument();
      expect(screen.getByText("level3")).toBeInTheDocument();
      expect(screen.getByText("level4")).toBeInTheDocument();
      expect(screen.getByText("level5")).toBeInTheDocument();
    });

    it("handles special characters in segments", () => {
      render(<Breadcrumb currentRoute="/products/t-shirts/blue" />);

      expect(screen.getByText("t-shirts")).toBeInTheDocument();
    });
  });

  // ============================================================================
  // INTEGRATION TESTS
  // ============================================================================

  describe("Props Integration", () => {
    it("spreads linkProps to Link components", () => {
      const handleClick = vi.fn();

      render(
        <Breadcrumb
          currentRoute="/products/shirts"
          linkProps={{
            onClick: handleClick,
            "data-testid": "breadcrumb-link",
          }}
        />
      );

      const links = screen.getAllByTestId("breadcrumb-link");
      expect(links.length).toBeGreaterThan(0);

      // Click first link
      links[0].click();
      expect(handleClick).toHaveBeenCalled();
    });

    it("uses custom startRoute and startRouteUrl", () => {
      render(
        <Breadcrumb
          currentRoute="/products"
          startRoute="Dashboard"
          startRouteUrl="/dashboard"
        />
      );

      const startLink = screen.getByRole("link", { name: "Dashboard" });
      expect(startLink).toBeInTheDocument();
      expect(startLink).toHaveAttribute("href", "/dashboard");
    });

    it("renders custom spacer element", () => {
      render(
        <Breadcrumb
          currentRoute="/products/shirts"
          spacer={<span data-testid="custom-spacer">→</span>}
        />
      );

      const spacers = screen.getAllByTestId("custom-spacer");
      expect(spacers.length).toBeGreaterThan(0);
      expect(spacers[0]).toHaveTextContent("→");
    });

    it("applies custom id to nav element", () => {
      render(
        <Breadcrumb
          currentRoute="/products"
          id="custom-breadcrumb"
        />
      );

      const nav = screen.getByRole("navigation");
      expect(nav).toHaveAttribute("id", "custom-breadcrumb");
    });

    it("applies custom className via classes prop", () => {
      render(
        <Breadcrumb
          currentRoute="/products"
          classes="custom-breadcrumb-class"
        />
      );

      const nav = screen.getByRole("navigation");
      expect(nav).toHaveClass("custom-breadcrumb-class");
    });

    it("applies custom inline styles", () => {
      render(
        <Breadcrumb
          currentRoute="/products"
          styles={{ padding: "1rem", backgroundColor: "lightgray" }}
        />
      );

      const nav = screen.getByRole("navigation");
      expect(nav).toHaveStyle({ padding: "1rem" });
    });
  });

  // ============================================================================
  // SUB-COMPONENT EXPORTS TESTS
  // ============================================================================

  describe("Sub-component Exports", () => {
    it("exports Nav sub-component", () => {
      expect(Breadcrumb.Nav).toBeDefined();
    });

    it("exports List sub-component", () => {
      expect(Breadcrumb.List).toBeDefined();
    });

    it("exports Item sub-component", () => {
      expect(Breadcrumb.Item).toBeDefined();
    });

    it("allows custom composition with sub-components", () => {
      render(
        <Breadcrumb.Nav aria-label="Custom breadcrumb">
          <Breadcrumb.Item>
            <a href="/">Home</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <span>Current Page</span>
          </Breadcrumb.Item>
        </Breadcrumb.Nav>
      );

      expect(screen.getByRole("navigation")).toBeInTheDocument();
      expect(screen.getByText("Home")).toBeInTheDocument();
      expect(screen.getByText("Current Page")).toBeInTheDocument();
    });
  });

  // ============================================================================
  // SNAPSHOT TESTS
  // ============================================================================

  describe("Snapshot Tests", () => {
    it("matches snapshot for simple breadcrumb", () => {
      const { container } = render(
        <Breadcrumb currentRoute="/products/shirts" />
      );
      expect(container).toMatchSnapshot();
    });

    it("matches snapshot with custom routes", () => {
      const routes: CustomRoute[] = [
        { path: "products", name: "All Products", url: "/products" },
      ];

      const { container } = render(
        <Breadcrumb currentRoute="/products/shirts" routes={routes} />
      );
      expect(container).toMatchSnapshot();
    });

    it("matches snapshot with truncation", () => {
      const { container } = render(
        <Breadcrumb
          currentRoute="/products/verylongproductname"
          truncateLength={10}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
