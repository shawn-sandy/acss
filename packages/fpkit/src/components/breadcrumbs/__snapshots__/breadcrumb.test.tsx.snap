// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > Snapshot Tests > matches snapshot for simple breadcrumb 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
  >
    <ol
      data-list="unstyled inline"
    >
      <li
        data-list="unstyled inline"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <a
          href="products"
        >
          products
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <span
          aria-current="page"
        >
          shirts
        </span>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumb > Snapshot Tests > matches snapshot with custom routes 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
  >
    <ol
      data-list="unstyled inline"
    >
      <li
        data-list="unstyled inline"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <a
          href="/products"
        >
          All Products
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <span
          aria-current="page"
        >
          shirts
        </span>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`Breadcrumb > Snapshot Tests > matches snapshot with truncation 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
  >
    <ol
      data-list="unstyled inline"
    >
      <li
        data-list="unstyled inline"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <a
          href="products"
        >
          products
        </a>
      </li>
      <li
        data-list="unstyled inline"
      >
        <span
          aria-hidden="true"
        >
          /
        </span>
        <span
          aria-current="page"
          aria-label="verylongproductname"
        >
          verylongpr...
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
